app/uk/gov/hmrc/apihubapplications/services/OASService.scala

```diff
@@ -16,15 +16,52 @@
 
 package uk.gov.hmrc.apihubapplications.services
 import com.google.inject.{Inject, Singleton}
+import io.circe.{yaml}
 import uk.gov.hmrc.apihubapplications.connectors.APIMConnector
-import uk.gov.hmrc.apihubapplications.models.apim._
+import uk.gov.hmrc.apihubapplications.models.apim.{Error as ApimError, *}
 import uk.gov.hmrc.apihubapplications.models.exception.ApimException
 import uk.gov.hmrc.http.HeaderCarrier
 
-import scala.concurrent.Future
+import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
 class OASService @Inject()(apimConnector: APIMConnector) {
-  def validateInPrimary(oas: String)(implicit hc: HeaderCarrier): Future[Either[ApimException, ValidateResponse]] =
-    apimConnector.validateInPrimary(oas)
+
+  val oasTitleError = ApimError("APIM", "Oas title is too long.")
+
+  def validateInPrimary(oas: String, validateTitle: Boolean)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Either[ApimException, ValidateResponse]] = {
+
+    if (validateTitle) {
+      val validOasTitle = isValidOasTitle(oas)
+      apimConnector.validateInPrimary(oas) map {
+        case Right(SuccessfulValidateResponse) if validOasTitle => Right(SuccessfulValidateResponse)
+        case Right(SuccessfulValidateResponse) => Right(InvalidOasResponse(newFailuresResponse))
+        case Right(invalidOasResponse: InvalidOasResponse) if validOasTitle => Right(invalidOasResponse)
+        case Right(invalidOasResponse: InvalidOasResponse) if invalidOasResponse.failure.errors.isDefined =>
+          val failure = invalidOasResponse.failure
+          val existingErrors = failure.errors.get
+          Right(invalidOasResponse.copy(failure = failure.copy(errors = Some(existingErrors ++ Seq(oasTitleError)))))
+        case Right(invalidOasResponse: InvalidOasResponse) =>
+          val failure = invalidOasResponse.failure
+          Right(invalidOasResponse.copy(failure = failure.copy(errors = Some(Seq(oasTitleError)))))
```

Comment:
I don't know if you'd like to simplify those two cases with `failure.errors.toList.flatten ++ Seq(oasTitleError)`
```suggestion
        case Right(invalidOasResponse: InvalidOasResponse) =>
          val failure = invalidOasResponse.failure
          Right(invalidOasResponse.copy(failure = failure.copy(
            errors = Some(failure.errors.toList.flatten ++ Seq(oasTitleError))))
          )
```